src/index.spec.ts

```diff
@@ -89,183 +88,310 @@ describe('EppoClient E2E test', () => {
-            break;
+            const na = client.getNumericAssignment(
+              subject.subjectKey,
+              experiment,
+              subject.subjectAttributes,
+            );
+            if (na === null) return null;
+            return na as number;
           }
           case ValueTestType.StringType: {
-            const stringAssignments = assignments.map((a) => a ?? null);
-            expect(stringAssignments).toEqual(expectedAssignments);
-            break;
+            const sa = client.getStringAssignment(
+              subject.subjectKey,
+              experiment,
+              subject.subjectAttributes,
+            );
+            if (sa === null) return null;
+            return sa as string;
           }
           case ValueTestType.JSONType: {
-            const jsonStringAssignments = assignments.map((a) => a ?? null);
-            expect(jsonStringAssignments).toEqual(expectedAssignments);
-            break;
+            const sa = client.getJSONStringAssignment(
+              subject.subjectKey,
+              experiment,
+              subject.subjectAttributes,
+            );
+            const oa = client.getParsedJSONAssignment(
+              subject.subjectKey,
+              experiment,
+              subject.subjectAttributes,
+            );
+            if (oa == null || sa === null) return null;
+            return sa as string;
           }
         }
-      },
-    );
+      });
+    }
   });
 
-  it('assigns subject from overrides when experiment is enabled', () => {
-    const mockConfigRequestor = td.object<ExperimentConfigurationRequestor>();
-    const mockPoller = td.object<IPoller>();
-    td.when(mockConfigRequestor.configStore.get(flagKey)).thenReturn({
-      ...mockExperimentConfig,
-      overrides: {
-        '1b50f33aef8f681a13f623963da967ed': 'variant-2',
-      },
-      typedOverrides: {
-        '1b50f33aef8f681a13f623963da967ed': 'variant-2',
+  describe('initialization errors', () => {
+    const maxRetryDelay = POLL_INTERVAL_MS * POLL_JITTER_PCT;
+    const mockConfigResponse = {
+      flags: {
+        [flagKey]: mockExperimentConfig,
       },
-    });
-    const client = new EppoClient(mockConfigRequestor, mockPoller);
-    const assignment = client.getAssignment('subject-10', flagKey);
-    expect(assignment).toEqual('variant-2');
-  });
+    };
 
-  it('assigns subject from overrides when experiment is not enabled', () => {
-    const mockConfigRequestor = td.object<ExperimentConfigurationRequestor>();
-    const mockPoller = td.object<IPoller>();
-    td.when(mockConfigRequestor.configStore.get(flagKey)).thenReturn({
-      ...mockExperimentConfig,
-      overrides: {
-        '1b50f33aef8f681a13f623963da967ed': 'variant-2',
-      },
-      typedOverrides: {
-        '1b50f33aef8f681a13f623963da967ed': 'variant-2',
-      },
+    it('retries initial configuration request before resolving', async () => {
+      td.replace(HttpClient.prototype, 'get');
```

Comment:
This (`HttpClient.get()`)was the easiest thing to hook into for throwing an error in the polling callback.